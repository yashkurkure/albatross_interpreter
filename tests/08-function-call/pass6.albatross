var i int := 0;
var j int := 0;

fun fs string () {
  return "";
}

fun fi int () {
  i := j;
  return i+j;
}

fun fv void () {
  i := 10;
}

fv();
fs();
fi();
return fi();
