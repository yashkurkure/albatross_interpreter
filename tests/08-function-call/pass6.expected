Variable declared "i" type int
Variable declared "j" type int
Function declared "fs" returns string
Function declared "fi" returns int
Variable read "j" type int
Variable written "i" type int
Variable read "i" type int
Variable read "j" type int
Function declared "fv" returns void
Variable written "i" type int
Function called "fv" returns void
Function called "fs" returns string
Function called "fi" returns int
Function called "fi" returns int
